### PR TITLE
MAKE-276: implement panic recovery for job execution

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: pool/group.go [hunk 1]
```diff
@@ -171,26 +172,46 @@ func (r *Group) Start(ctx context.Context) error {
...
+		// if we hit a panic we want to add an error to the job;
+		err = recovery.HandlePanicWithError(recover(), nil, "worker process encountered error")
+		if err != nil {
+			if unit.j != nil {
+				unit.j.AddError(err)
+				unit.q.Complete(ctx, unit.j)
+			}
+			// start a replacement worker.
+			wg.Add(1)
+			go groupWorker(ctx, wg, name, work)
```
Comment (bsamek) [inline pool/group.go]: Couldn't this continue infinitely, each groupWorker panicking and spawning a new new one?

File: pool/group.go [hunk 2]
```diff
@@ -171,26 +172,46 @@ func (r *Group) Start(ctx context.Context) error {
...
 	return nil
 }
 
+func groupWorker(ctx context.Context, wg *sync.WaitGroup, name string, work <-chan workUnit) {
+	var (
+		err  error
+		unit workUnit
+	)
+
+	defer wg.Done()
```
Comment (bsamek) [inline pool/group.go]: Where is the Wait?

Comment (tychoish) [reply pool/group.go]: in the close function for the pool

File: pool/group.go (hunk 1 above)
Comment (tychoish) [reply pool/group.go]: we run this only when there's a panic (e.g. 190)

**Pull request merged**